### PR TITLE
fix: handle registration attempts with an already-registered email

### PR DESCRIPTION
**Pull request by mlnps**
## Summary
- Fixes registration for already-registered emails: previously, submitting the registration form again with an existing email showed the same generic "registrationn early done, check your email" success screen, but nothing actually happened, no email was ever sent, regardless of whether the existing account was confirmed or not. Users had no way to realize they'd already registered.
- New `POST /auth/register` backend endpoint owns the full registration flow: new email -> create account, existing + confirmed -> send a password reset email, existing + unconfirmed -> resend the confirmation email. The response is identical in all cases, so, as a deliberate side effect of fixing this properly, it also never reveals to the caller which branch fired.
- Frontend registration + resend now go through this endpoint instead of calling Supabase directly.

## Why backend-owned
Deciding what should actually happen (send a reset vs. resend a confirmation) requires knowing whether the email is already registered and confirmed. That lookup requires the service-role client, which only exists server-side. The frontend calling Supabase directly had no way to make that distinction, which is why it silently did nothing useful before.

## Also included
The "check your email" screen now shows the actual address the message was sent to, and (main frontend only) a "Falsche E-Mail-Adresse? Erneut registrieren" link to go back and re-register. Previously, once this screen appeared there was no way back to the form at all, a pre-existing gap in `AuthLayout`'s gating that this PR happens to make fixable.

## Summary by CodeRabbit

- **New Features**
  - Added a unified registration and account-recovery flow for new, confirmed, and unconfirmed email addresses.
  - Users can reset passwords or resend confirmation emails through the same flow.
- **Bug Fixes**
  - Kept messaging generic for existing accounts to avoid revealing account status.
- **Content**
  - Improved unconfirmed-email guidance and wrong-email navigation.
  - Expanded German password-recovery email instructions.
- **Tests**
  - Expanded coverage for registration, recovery, confirmation resending, and the unified success screen.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1215791302655155

### CODE REVIEW

File: apps/backend/src/index.ts [hunk 1]
```diff
@@ -8,6 +8,7 @@ import documents from "./routes/documents";
 import llms from "./routes/llms";
 import { config, verifyConfig } from "./config";
 import admin from "./routes/admin";
+import auth from "./routes/auth";
```
[the remark this answers is not in the document]
Comment (mlnps) [reply apps/backend/src/index.ts:11]: basicAuth requires a valid Supabase JWT session token. A user registering has no session yet, by defintion. True about sentry logging though, will change this.

File: apps/backend/src/integration/routes/auth.integration.test.ts [hunk 2]
```diff
@@ -0,0 +1,214 @@
...
+				)
+			) {
+				continue;
+			}
+			await serviceRoleDbClient.auth.admin.deleteUser(user.id);
+		}
+	}
+
+	beforeAll(async () => {
+		await deleteTestUsersIfAny();
```
Comment (raphael-arce) [inline apps/backend/src/integration/routes/auth.integration.test.ts]: I think we should avoid cleaning up before a test: this implies a cleanup after a test is flaky.

File: apps/backend/src/integration/routes/auth.integration.test.ts [hunk 3]
```diff
@@ -0,0 +1,214 @@
...
+import { serviceRoleDbClient } from "../../supabase";
+
+describe("/auth/register", () => {
+	const givenNewEmail = "auth-register-test-suite-new@ts.berlin";
+	const givenConfirmedEmail = "auth-register-test-suite-confirmed@ts.berlin";
+	const givenUnconfirmedEmail =
+		"auth-register-test-suite-unconfirmed@ts.berlin";
+	const givenPassword = "SecurePassword123!";
+
+	const userIds: Record<string, string> = {};
```
Comment (raphael-arce) [inline apps/backend/src/integration/routes/auth.integration.test.ts]: you can pass a pre-defined id during user creation, which allows to you avoid having to store user ids after creation to be able to use them for deletion later on 👍

File: apps/backend/src/integration/routes/auth.integration.test.ts [hunk 4]
```diff
@@ -0,0 +1,214 @@
...
+
+	async function deleteTestUsersIfAny(): Promise<void> {
+		const { data, error } = await serviceRoleDbClient.auth.admin.listUsers();
```
Comment (raphael-arce) [inline apps/backend/src/integration/routes/auth.integration.test.ts]: There is an edge case here: this call has a limit of 50 users by default, the users we're looking for might not be included in the response but still be there. I'd suggest to rather directly delete the users via pre-defined ID 👍

File: apps/backend/src/services/registration-service.ts [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+		password,
+		firstName,
+		lastName,
+	}: {
+		email: string;
+		password?: string;
+		firstName?: string;
+		lastName?: string;
+	}): Promise<void> {
+		try {
```
Comment (raphael-arce) [inline apps/backend/src/services/registration-service.ts]: I think there is no need to wrap the whole function in a try catch, as all called functions return errors instead of throwing them. Would suggest to just captureError + return instead of throw + catch locally.

File: apps/backend/supabase/migrations/20260728140000_create_check_email_registration_status_function.sql [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+  v_found := FOUND;
+
+  RETURN QUERY SELECT v_found, (v_found AND v_confirmed_at IS NOT NULL);
+END;
+$$;
+
+REVOKE
+EXECUTE ON FUNCTION public.check_email_registration_status (TEXT)
+FROM
+    PUBLIC;
```
Comment (raphael-arce) [inline apps/backend/supabase/migrations/20260728140000_create_check_email_registration_status_function.sql:26]: I think we also need to revoke from `anon` and `authenticated`, otherwise everybody can call this rpc, e.g. like this:
```ts
import { supabaseAnonClient } from "./tests/supabase";

async function test() {
	const { data, error } = await supabaseAnonClient.rpc(
		"check_email_registration_status",
		{
			p_email: "<REGISTERED-EMAIL>",
		},
	);

	console.log({ data, error }); 
}

test(); 
```

File: apps/frontend/src/store/auth-store.ts [hunk 7]
```diff
@@ -143,9 +181,13 @@ export const useAuthStore = create<AuthStore>()((set, get) => {
 		}
 	});
 
+	const storedUnconfirmedEmail = readUnconfirmedEmail();
```
Comment (raphael-arce) [inline apps/frontend/src/store/auth-store.ts]: I think storing the unconfirmed email in the session is not ideal. When I try to register, I get to the unconfirmed page, but when I try to register again or to log-in I'm trapped on the unconfirmed page. Would rather keep that state in memory which would reset itself on page leave / refresh.

File: apps/frontend/src/store/auth-store.ts [hunk 8]
```diff
@@ -233,9 +278,9 @@ export const useAuthStore = create<AuthStore>()((set, get) => {
 
 			resendTime = Date.now();
 
-			const { error } = await resendEmailConfirmation(unconfirmedEmail);
-
-			if (error) {
+			try {
+				await registerUser({ email: unconfirmedEmail });
```
Comment (raphael-arce) [inline apps/frontend/src/store/auth-store.ts]: The naming was confusing me a bit here. Would suggest to name the frontend api function something like `registerOrRecoverUser`, similar to the function in the backend service 👍

**Review by raphael-arce — COMMENTED** [review]
The edge-cases are quite complex, they're all managed as fas as I can tell, well done! I have a few comments though 👍

File: apps/backend/src/integration/routes/auth.integration.test.ts (hunk 4 above)
Comment (mlnps) [reply apps/backend/src/integration/routes/auth.integration.test.ts]: Good call!

File: apps/backend/src/integration/routes/auth.integration.test.ts (hunk 2 above)
Comment (mlnps) [reply apps/backend/src/integration/routes/auth.integration.test.ts]: I see that it could mask a broken afterAll, I just sometimes had the issues in development when tests did not run through completely because some process got killed, I had leftovers which then the next time made problems. But you're right that this is an edge case and normally afterAll should to the job. I don't know. I don't have a strong opinion here.

File: apps/backend/src/integration/routes/auth.integration.test.ts (hunk 3 above)
Comment (mlnps) [reply apps/backend/src/integration/routes/auth.integration.test.ts]: Yes, you are right, that's nicer. Although we can only do that for users we create via createUser(), but not for the ones hitting the real /auth/register endpoint (not returning the id by design, so its identical for every branch).

File: apps/backend/src/services/registration-service.ts (hunk 5 above)
Comment (mlnps) [reply apps/backend/src/services/registration-service.ts]: The reason for this was to make sure that, no matter what, even in case of a network failure or an unexcpected excpetion (which actually would be thrown from the three GoTrue functions signUp(), resend() and resetPasswordForEmail(), since the only returning known errors), this sensitive endpoint always returns {"message: "ok"} to any potential caller.  But I agree that currently I can't think of a concrete scenario where this could currently be explited. It's just an extra security layer on top.

File: apps/backend/supabase/migrations/20260728140000_create_check_email_registration_status_function.sql (hunk 6 above)
Comment (mlnps) [reply apps/backend/supabase/migrations/20260728140000_create_check_email_registration_status_function.sql:26]: oooha yes, thanks

File: apps/frontend/src/store/auth-store.ts (hunk 7 above)
Comment (mlnps) [reply apps/frontend/src/store/auth-store.ts]: Yes definitely. For some weird reason I thought it was done like that before already, can't comprehend now. changed it!

File: apps/frontend/src/store/auth-store.ts [hunk 9]
```diff
@@ -157,26 +157,23 @@ export const useAuthStore = create<AuthStore>()((set, get) => {
...
-			const { data, error } = await registerUser({
-				email,
-				password,
-				firstName,
-				lastName,
-			});
-
-			if (error) {
+			try {
+				await registerOrRecoverUser({ email, password, firstName, lastName });
```
[the remark this answers is not in the document]
Comment (mlnps) [reply apps/frontend/src/store/auth-store.ts:161]: This is intentional, not a bug. the whole point of this endpoint is that the response is always identical no matter what happens internally, so it can never leak whether an email is registered. surfacing failures differently, even generically, reintroduces exactly that discrepancy. Internal failures are captured via Sentry.

File: apps/backend/src/services/registration-service.ts (hunk 5 above)
Comment (raphael-arce) [reply apps/backend/src/services/registration-service.ts]: I understand your intent. I'd say unexpected errors are 5xx class errors and are already handled via the `.onError` middleware registered [here](https://github.com/technologiestiftung/baergpt/blob/main/apps/backend/src/index.ts#L44). I'd rather be in favor of letting the user know if something unexpected happened and that no email will be sent, so that they don't expect something that will never happen. I don't think we'll reveal anything worth concealing 👍

**Review by raphael-arce — APPROVED** [review]
one more thing then good to go! pre-approving 👍

**Review by raphael-arce — APPROVED** [review]
lgtm 👍
